cmd/tracee-ebpf/flags/filter.go

```diff
@@ -132,7 +132,7 @@ func PrepareFilter(filtersArr []string) (tracee.Filter, error) {
 			Filters: make(map[events.ID]filters.IntFilter),
 		},
 		ArgFilter: &filters.ArgFilter{
-			Filters: make(map[events.ID]map[string]filters.ArgFilterVal),
+			Filters: make(map[events.ID]map[string]filters.StringFilter),
```

Comment:
Today we do argument comparisons by converting to string first, but in the future, we might perform comparisons of other types (e.g. int), which will also be more efficient. For this reason the original implementation defined a map value of its own (ArgFilterVal), so why changing this here?

Reply:
Well, we're moving it to ebpf later anyway. I didn't see a use in having them separated since they are currently just the same, if we decide for some reason not to move this logic to ebpf anyway, and further improve it in userspace we would probably need to create a custom struct for it anyway. Otherwise it would just be duplication from the current StringFilter.

As a compromise I can change it to a separate struct which will embed the StringFilter like so:
```
type eventArgFilter struct {
    StringFilter
}
```

Reply:
Not sure if you thumbs upped for embedding or keeping as is

Reply:
Thumbs up for the compromise :-)